relay-profiling/src/sample.rs

```diff
@@ -95,6 +123,107 @@ impl Profile {
             frame.strip_pointer_authentication_code(addr);
         }
     }
+
+    fn remove_idle_samples_at_the_edge(&mut self) {
+        let mut active_ranges: HashMap<u64, Range<usize>> = HashMap::new();
```

Comment:
```suggestion
        let mut active_ranges = HashMap::new();
```
I think this should also work.

Reply:
Rust complains that the type must be known before hand.